api/app/models/fetch/predictions.py

```diff
@@ -192,6 +190,68 @@ async def _fetch_model_predictions_by_station_codes(model: ModelEnum, station_co
 
 
 async def fetch_model_predictions(model: ModelEnum, station_codes: List[int]):
-    """ Fetch 10 day global model weather predictions for a given station."""
+    """ Fetch model weather predictions for a given list of stations and a given model. """
     # Fetch predictions from the database.
     return await _fetch_model_predictions_by_station_codes(model, station_codes)
+
+
+async def _fetch_most_recent_historic_predictions_by_station_codes(model: ModelEnum, station_codes: List[int]) -> List[WeatherModelPrediction]:
+    """ Fetch the most recent historic model predictions from database based on each station's coordinates. """
+    # construct temporary helper dictionary of relevant weather stations
+    stations_dict = {}
+    stations = await app.stations.get_stations_by_codes(station_codes)
+    for station in stations:
+        stations_dict[station.code] = station
+
+    # construct helper dictionary of WeatherModelPredictions
+    # weather_model_predictions_dict is indexed by station_code, then by prediction_timestamp
+    weather_model_predictions_dict = defaultdict(dict)
+
+    # We're only interested in the last 5 days
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+    five_days_ago = now - datetime.timedelta(days=5)
+    # send the query
+    session = app.db.database.get_session()
+    historic_predictions = app.db.crud.get_historic_station_model_predictions(
+        session, station_codes, model, five_days_ago, now)
+    print(historic_predictions)
+    for prediction, prediction_model_run_timestamp, prediction_model in historic_predictions:
+        station_predictions = weather_model_predictions_dict.get(prediction.station_code)
+        existing_prediction = None
+        if station_predictions is not None:
```

Comment:
station_predictions is always going to be "not None" because weather_model_predictions_dict will always return a dict due to `weather_model_predictions_dict = defaultdict(dict)`

Reply:
station_predictions will be None if there's no value for the prediction.station_code key in weather_model_predictions_dict

Reply:
Duh - sorry - brain fail.